src/gui/sdlmain.cpp

```diff
@@ -1846,19 +1950,43 @@ static void SetupWindowResolution(const char *val)
 		SDL_Rect bounds;
 		SDL_GetDisplayBounds(sdl.display_number, &bounds);
 		if (w > 0 && h > 0 && w <= bounds.w && h <= bounds.h) {
+			sdl.desktop.want_resizable_window = false;
 			sdl.desktop.window.use_original_size = false;
 			sdl.desktop.window.width = w;
 			sdl.desktop.window.height = h;
 			return;
 		}
 	}
 
-	LOG_MSG("MAIN: 'windowresolution = %s' is not a valid setting, using "
-	        "'original' instead",
-	        pref.c_str());
+        LOG_MSG("MAIN: 'windowresolution = %s' is not a valid setting, using "
+                "'original' instead", pref.c_str());
+        sdl.desktop.want_resizable_window = false;
 	sdl.desktop.window.use_original_size = true;
 }
 
+static SDL_Rect CalculateViewport(int win_width, int win_height)
+{
+	const double prog_aspect_ratio = (sdl.draw.width * sdl.draw.scalex) /
+	                                 (sdl.draw.height * sdl.draw.scaley);
+	const double win_aspect_ratio = double(win_width) / double(win_height);
```

Comment:
Super minor.. could assert that draw.height, draw.scaley, and win_height aren't zero to prevent divide-by-zero (although this scenario might be possible depending on prior checks)

Reply:
I added asserts for draw.* and scale*.

Initially I thought there's no need for assert for win_height (as there's no divide-by-zero error for IEEE floating-point numbers), but I need to rethink it. I probably will end up adding this assert as well… (or maybe early return? not sure yet)

Reply:
OK, there's no need, asserts for draw and scale are enough. If win_height is 0, then `win_aspect_ratio` will be either NaN or +Inf; in both cases comparison will end up in `else` block and return value will be {0, 0, 0, 0}.